src/marks/raster.js

```diff
@@ -337,14 +322,91 @@ export function interpolatorBarycentric({random = randomLcg(42)} = {}) {
           if (gb < 0) continue;
           const gc = 1 - ga - gb;
           if (gc < 0) continue;
-          W[x + width * y] = mix(va, ga, vb, gb, vc, gc, x, y);
+          const i = x + width * y;
+          W[i] = mix(va, ga, vb, gb, vc, gc, x, y);
+          S[i] = 1;
         }
       }
     }
+    extrapolateBarycentric(W, S, X, Y, V, width, height, hull, index, mix);
     return W;
   };
 }
 
+// Extrapolate by finding the closest point on the hull.
+function extrapolateBarycentric(W, I, X, Y, V, width, height, hull, index, mix) {
```

Comment:
You’re still using `I` here instead of `S` (the earlier confusion I mentioned).

```suggestion
function extrapolateBarycentric(W, S, X, Y, V, width, height, hull, index, mix) {
```